src/main/java/hudson/plugins/cppncss/CppNCSSPublisher.java

```diff
@@ -80,6 +90,29 @@ protected Ghostwriter newGhostwriter() {
         return new CppNCSSGhostwriter(reportFilenamePattern, functionCcnViolationThreshold, functionNcssViolationThreshold, targets);
     }
 
+    @Override
+    public void perform(Run<?,?> run, FilePath workspace, Launcher launcher, TaskListener listener) {
+    	CppNCSSProjectIndividualReport report = new CppNCSSProjectIndividualReport(run.getParent(), functionCcnViolationThreshold, functionNcssViolationThreshold);
+    	ActionGetter getter = new ActionGetter();
+    	getter.addProjectAction(report);
+    	run.addAction(getter);
+        try {
+			BuildProxy.doPerform(newGhostwriter(), run, workspace, listener);
+		} catch (IOException | InterruptedException e) {
+			run.setResult(Result.FAILURE);
+			e.printStackTrace(listener.getLogger());
+		}
+    }
+    
+    @Override
+    public boolean perform(AbstractBuild<?,?> build, Launcher launcher, BuildListener listener) throws InterruptedException, java.io.IOException {
+    	FilePath wkspace = build.getWorkspace();
+    	if (wkspace == null)
+    		return false;
+    	perform(build, wkspace, launcher, listener);
+        return true;
+    }
+
     @Extension
     public static final class DescriptorImpl extends BuildStepDescriptor<Publisher> {

```

Comment:
Add `@Symbol("cppncss")` so that there is a good alias?

Reply:
Ok